test/form_builder/general_test.rb

```diff
@@ -117,8 +117,19 @@ def with_custom_form_for(object, *args, &block)
     assert_select 'form input#user_name.string'
   end
 
-  test 'builder generates text areas for text columns' do
+  test 'builder generates text field for text columns' do
     with_form_for @user, :description
+    if @user.respond_to?(:type_for_attribute) &&
+       @user.type_for_attribute('description').type == :string
+      assert_select 'form input#user_description.string'
```

Comment:
I think that only our test stubs are using `:string` instead of `:text` because `ActiveModel::Type::Text` no longer exists. We could backport it though:

```ruby
    begin
      ActiveModel::Type.lookup(:text)
    rescue ArgumentError
      class ::ActiveModel::Type::Text < ActiveModel::Type::String
        def type; :text; end
      end
      ActiveModel::Type.register(:text, ActiveModel::Type::Text)
    end
```